src/ContainerCheckpointModal.jsx

```diff
@@ -1,89 +1,70 @@
-                               onClick={() => {
-                                   this.handleCheckpointContainer({
-                                       keep: this.state.keep,
-                                       leaveRunning: this.state.leaveRunning,
-                                       tcpEstablished: this.state.tcpEstablished,
-                                       ignoreRootFS: this.state.ignoreRootFS
-                                   });
-                               }}>
-                           {_("Checkpoint")}
-                       </Button>
-                       <Button variant="link" isDisabled={this.state.inProgress}
-                               onClick={Dialogs.close}>
-                           {_("Cancel")}
-                       </Button>
-                   </>}
-            >
-                <Form isHorizontal>
-                    <Checkbox label={_("Keep all temporary checkpoint files")} id="checkpoint-dialog-keep"
-                                  name="keep" isChecked={this.state.keep} onChange={this.handleChange} />
-                    <Checkbox label={_("Leave running after writing checkpoint to disk")}
-                                  id="checkpoint-dialog-leaveRunning" name="leaveRunning"
-                                  isChecked={this.state.leaveRunning} onChange={this.handleChange} />
-                    <Checkbox label={_("Support preserving established TCP connections")}
-                                  id="checkpoint-dialog-tcpEstablished" name="tcpEstablished"
-                                  isChecked={this.state.tcpEstablished} onChange={this.handleChange} />
-                    <Checkbox label={_("Do not include root file-system changes when exporting")}
-                                  id="checkpoint-dialog-ignoreRootFS" name="ignoreRootFS"
-                                  isChecked={this.state.ignoreRootFS} onChange={this.handleChange} />
-                </Form>
-            </Modal>
-        );
-    }
-}
+    return (
+        <Modal isOpen
+               showClose={false}
+               position="top" variant="medium"
+               title={cockpit.format(_("Checkpoint container $0"), containerWillCheckpoint.Names)}
+               footer={<>
+                   <Button variant="primary" isDisabled={inProgress}
+                           isLoading={inProgress}
+                           onClick={handleCheckpointContainer}>
+                       {_("Checkpoint")}
+                   </Button>
+                   <Button variant="link" isDisabled={inProgress}
+                           onClick={Dialogs.close}>
+                       {_("Cancel")}
+                   </Button>
+               </>}
+        >
+            <Form isHorizontal>
+                <Checkbox label={_("Keep all temporary checkpoint files")} id="checkpoint-dialog-keep"
+                              name="keep" isChecked={keep} onChange={(checked, _) => setKeep(checked)} />
```

Comment:
This can be more nicely written by just using the function name how it was before like:
```
onChange(setKeep)
```

Same for all the bellow.

Reply:
Fixed it everywhere, it's nicer and cleaner!